libsolidity/analysis/TypeChecker.cpp

```diff
@@ -450,6 +450,9 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	TypePointer varType = _variable.annotation().type;
 	solAssert(!!varType, "Variable type not provided.");
 
+	if (auto contractType = dynamic_cast<ContractType const*>(varType))
+		if (contractType->contractDefinition().isLibrary())
+			m_errorReporter.typeError(_variable.location(), "Variables cannot be of type Library.");
```

Comment:
```suggestion
			m_errorReporter.typeError(_variable.location(), "The type of a variable cannot be a library.");
```
(the type itself is not "library")